gitclone/checkout.go

```diff
@@ -277,12 +294,18 @@ func createCheckoutStrategy(checkoutMethod CheckoutMethod, cfg Config, patchFile
 }
 
 func selectFetchOptions(method CheckoutMethod, cloneDepth int, fetchTags, fetchSubmodules bool, filterTree bool) fetchOptions {
+	// If cloneDepth is 0, that means the user did not set a value for it,
+	// so we will determine the correct value based on the checkout method.
+	if cloneDepth == 0 {
+		cloneDepth = idealDefaultCloneDepth(method)
```

Comment:
Hey folks, I wonder what this code is doing here. It's definitely wasn't present in 6.2.2 this PR claims to revert to.

Reply:
Oh never mind, sorry, it reverts the revert 🤦